File: androidchallenge1/src/main/java/class28/androidchallenge01/ui/platesdetails/ActPlateDetails.java
package class28.androidchallenge01.ui.platesdetails;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import class28.androidchallenge01.R;
import class28.androidchallenge01.model.Plate;

import static class28.androidchallenge01.ui.plates.ActRestaurantPlates.CHOSEN_PLATE;

public class ActPlateDetails extends AppCompatActivity {

    ImageButton plateDetailsBackButton;
    ImageView plateDetailsImage;
    TextView plateDetailsName, plateDescription;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_plate_details);
        plateDetailsBackButton = findViewById(R.id.plate_details_back);
        plateDetailsImage = findViewById(R.id.plate_details_image);
        plateDetailsName = findViewById(R.id.plate_details_name);
        plateDescription = findViewById(R.id.plate_description);

        plateDetailsBackButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });

        Plate plate = getIntent().getParcelableExtra(CHOSEN_PLATE);
        plateDetailsImage.setImageResource(plate.getPlateImage());
        plateDetailsName.setText(plate.getName());
        plateDescription.setText(plate.getDescription());
    }
}

File: androidchallenge1kotlin/src/main/java/class28/androidchallenge01kotlin/ui/platesdetails/ActPlateDetails.kt
package class28.androidchallenge01kotlin.ui.platesdetails

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import class28.androidchallenge01kotlin.R
import class28.androidchallenge01kotlin.model.Plate
import class28.androidchallenge01kotlin.ui.plates.ActRestaurantPlates.Companion.CHOSEN_PLATE
import kotlinx.android.synthetic.main.activity_plate_details.*

class ActPlateDetails : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_plate_details)

        plate_details_back.setOnClickListener { onBackPressed() }

        val plate = intent.getParcelableExtra(CHOSEN_PLATE) as Plate

        plate_details_image.setImageResource(plate.plateImageRes)
        plate_details_name.text = plate.name
        plate_description.text = plate.description
    }
}